Crash in rule creator suppressed in `npm test`

This was discovered in analysis of a failure in #5349.

In latest master (2bd57b6bbd11b23fdec65dc11c691f74c6a8db75) on any version of Node, if:

- A rule creator throws an exception (e.g. while parsing options)
- That code path is one not used in ESLint's own config (e.g. `max-statements`, which we don't use at all)

...then `npm test` won't show any information about the failure:

```
> eslint@2.2.0 test /Users/brandon/code/eslint/eslint
> node Makefile.js test

Validating Makefile.js
Validating JSON Files
Validating Markdown Files
Validating JavaScript files
Validating JavaScript test files
Validating rules

  [▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬]=============================================================================
Writing coverage object [/Users/brandon/code/eslint/eslint/coverage/coverage.json]
Writing coverage reports at [/Users/brandon/code/eslint/eslint/coverage]
=============================================================================


  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
  ․․․․․

  215 passing (560ms)

npm ERR! Test failed.  See above for more details.
```

You can try this yourself with the following diff, which will throw an error silently when you run `npm test`:

```diff
diff --git a/lib/rules/max-statements.js b/lib/rules/max-statements.js
index 78de52e..1e37c30 100644
--- a/lib/rules/max-statements.js
+++ b/lib/rules/max-statements.js
@@ -12,6 +12,8 @@

 module.exports = function(context) {

+    process.thisDoesntExist();
+
     //--------------------------------------------------------------------------
     // Helpers
     //--------------------------------------------------------------------------
```

If the error is thrown by a configuration that ESLint itself uses, then the stack trace will be shown as part of the "Validating JavaScript files" step. Applying the above diff to `eqeqeq`, for example, would fail with a stack trace before it ever got to the rule tests.

--- a/tests/lib/config/config-initializer.js
+++ b/tests/lib/config/config-initializer.js
@@ -220,8 +220,15 @@ describe("configInitializer", function() {
                 origLog = console.log;
                 console.log = function() {}; // necessary to replace, because of progress bar
                 process.chdir(fixtureDir);
-                config = init.processAnswers(answers);
-                process.chdir(originalDir);
+                try {
+                    config = init.processAnswers(answers);
+                    process.chdir(originalDir);
+                } catch (err) {
+                    // if processAnswers crashes, we need to be sure to restore cwd and console.log
+                    console.log = origLog;
+                    process.chdir(originalDir);
+                    throw err;
+                }
             });
 
             beforeEach(function() {

